### PR TITLE
Make changes required by changes in aws-nitro-enclaves-nsm-api.

### PR DESCRIPTION
**Pull request by egrimley-arm**
They have reorganised the subcrates into modules.

Redirect to egrimley-arm to test while ring is not yet updated.

### CODE REVIEW

File: session-manager/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,6 @@ std = [
 [dependencies]
 err-derive = "0.2"
 policy-utils = { path = "../policy-utils" }
-ring = { git = "https://github.com/veracruz-project/ring.git", branch = "veracruz" }
-rustls = { git = "https://github.com/veracruz-project/rustls.git", branch = "veracruz" }
-webpki = { git = "https://github.com/veracruz-project/webpki.git", branch = "veracruz" }
+ring = { git = "https://github.com/egrimley-arm/ring.git", branch = "veracruz" }
+rustls = { git = "https://github.com/egrimley-arm/rustls.git", branch = "veracruz" }
```
Comment (dreemkiller) [inline session-manager/Cargo.toml:29]: Update this after the rustls PR goes in, or revert if there isn't one

File: veracruz-server-test/Cargo.toml [hunk 2]
```diff
@@ -36,12 +36,12 @@ lazy_static = "1.4"
 log = "=0.4.13"
 policy-utils = { path = "../policy-utils", optional = true }
 proxy-attestation-server = { path = "../proxy-attestation-server" }
-ring = { git = "https://github.com/veracruz-project/ring.git", branch = "veracruz" }
-rustls = { git = "https://github.com/veracruz-project/rustls.git", branch = "veracruz" }
+ring = { git = "https://github.com/egrimley-arm/ring.git", branch = "veracruz" }
```
Comment (dreemkiller) [inline veracruz-server-test/Cargo.toml:39]: Update this after https://github.com/veracruz-project/ring/pull/4 goes in

**Review by dreemkiller — CHANGES_REQUESTED** [review]
Wow! Quick work on this one.
There's about 35 places in `Cargo.toml`'s that need to be updated when https://github.com/veracruz-project/ring/pull/4 goes in (I've already approved it), or need to be reverted if we don't need changes for `rustls` or `webpki`.

**Comment by mathias-arm** [comment]
@dreemkiller I think what @egrimley-arm intended is to remove [533e469](https://github.com/veracruz-project/veracruz/pull/345/commits/533e469c05e006dce281dfc6bd8d5ea1ce6ca1d6) from the PR once https://github.com/veracruz-project/ring/pull/4 is merged.

This is another example of why #164 is needed. The equivalent to this PR on that branch is [9204430](https://github.com/veracruz-project/veracruz/pull/164/commits/92044308a2d0ac1c03527da0f11b1058384c895b).

**Comment by egrimley-arm** [comment]
> This is another example of why #164 is needed. The equivalent to this PR on that branch is [9204430](https://github.com/veracruz-project/veracruz/pull/164/commits/92044308a2d0ac1c03527da0f11b1058384c895b).

Yes, I had to fork and modify the transitive closure of the repos that use `ring`. Some of that can be automated, with `perl -i`, but the actual forking requires browser interaction with GitHub.

**Comment by egrimley-arm** [comment]
I've also removed a couple of version pinnings (`version = "=0.16.11"`). I had to remove them for testing, and they don't seem to be needed, so it didn't seem helpful to put them back!

File: proxy-attestation-server/Cargo.toml [hunk 3]
```diff
@@ -49,13 +49,13 @@ http = "=0.2.4"
 lazy_static = "1.3"
 log = "=0.4.13"
 nb-connect = "=1.0.3"
-nitro-enclave-attestation-document = { git = "https://github.com/veracruz-project/nitro-enclave-attestation-document.git", rev = "4d703bcde209a8d625776f6c346c36a9ad084511", optional = true }
+nitro-enclave-attestation-document = { git = "https://github.com/egrimley-arm/nitro-enclave-attestation-document.git", branch = "rename", optional = true }
```
[the remark this answers is not in the document]
Comment (egrimley-arm) [reply proxy-attestation-server/Cargo.toml]: No, it was only required by transitivity.

File: session-manager/Cargo.toml [hunk 4]
```diff
@@ -25,6 +25,6 @@ std = [
...
+webpki = { git = "https://github.com/egrimley-arm/webpki.git", branch = "veracruz" }
```
[the remark this answers is not in the document]
Comment (egrimley-arm) [reply session-manager/Cargo.toml]: No, it was just required by transitivity.

File: veracruz-client/Cargo.toml [hunk 5]
```diff
@@ -36,14 +36,14 @@ openssl-sys = "=0.9.70"
...
+ring = { git = "https://github.com/egrimley-arm/ring.git", branch = "veracruz" }
 # The cargo patch mechanism does NOT work when we add function into a macro_rules!
-rustls = { git = "https://github.com/veracruz-project/rustls.git", branch = "veracruz" }
+rustls = { git = "https://github.com/egrimley-arm/rustls.git", branch = "veracruz" }
 serde_json = { git = "https://github.com/veracruz-project/json.git", branch = "veracruz" }
 structopt = { version = "0.3", optional = true, features = ["wrap_help"] }
 transport-protocol = { path = "../transport-protocol" }
 veracruz-utils = { path = "../veracruz-utils", features = ["std"] }
-webpki = { git = "https://github.com/veracruz-project/webpki.git", branch = "veracruz" }
+webpki = { git = "https://github.com/egrimley-arm/webpki.git", branch = "veracruz" }
```
[the remark this answers is not in the document]
Comment (egrimley-arm) [reply veracruz-client/Cargo.toml]: I don't think so.

**Review by gbryant-arm — APPROVED** [review]

**Pull request merged**